### ISSUE TITLE
QPSK phase correction is wrong

### ISSUE THREAD

**Issue by Digitelektro** [artlav/meteor_decoder#12]
I think the phase correction in the fix_packet function is not correct. I was wondering why your implementation gives much poorer quality images compared to the LRPT decoder from Robonuka. I am implementing my own Meteor M2 decoder, and I researched a lot how it works and what could be the problem. There is no write access to your repository, so here is what I've changed in your code, it gives much better images now.

```procedure fix_packet(data:pointer;len:integer;shift:integer);
var j:integer;
d:pshortinta;
b:shortint;
begin
 d:=data;
 case shift of
  4:for j:=0 to len div 2-1 do begin
   b:=d[j*2+0];
   d[j*2+0]:= -d[j*2+1];
   d[j*2+1]:= -b;
  end;
  5:for j:=0 to len div 2-1 do begin
   d[j*2+0]:=-d[j*2+0];
  end;
  6:for j:=0 to len div 2-1 do begin
   b:=d[j*2+0];
   d[j*2+0]:= d[j*2+1];
   d[j*2+1]:= b;
  end;
  7:for j:=0 to len div 2-1 do begin
   d[j*2+1]:=-d[j*2+1];
  end;
 end;
end;```

**Comment by pfranchini** [comment]
> There is no write access to your repository,

In github you need to do a "pull request" from your own fork of the code. Or the author might just want to do himself the changes.

**Comment by Digitelektro** [comment]
Okay, thank you! I've done it.

**Comment by pfranchini** [comment]
thank you, very curious to try your bug fix. Are you writing some linux/c++ code that could be used from command line? Keep us informed, please.

**Comment by Digitelektro** [comment]
Yes, I am writing a cross platform program both for Linux and Windows. It will have a lot of extra feature than just decoding the images. It will correct the missing lines from the image and it will produce images projected onto a map. I have all the working algorithms now in C# (just for simplicity), I just have to port it to C++. It took months for me to understand the math and everything behind it, but finally I have a working solution. I will upload soon the first version as soon as my time allows to get it ready.

**Comment by pfranchini** [comment]
Sounds really interesting, so far I could not find anything like that for linux. Will follow your repo. Thanks

**Comment by Digitelektro** [comment]
Hi @pfranchini 
I totally forget to update you about the software I mentioned, I accidentally bumped into this ticket now, so it is here: https://github.com/Digitelektro/MeteorDemod
